server/src/main/java/org/apache/druid/server/coordinator/ServerHolder.java

```diff
@@ -20,31 +20,61 @@
 package org.apache.druid.server.coordinator;
 
 import org.apache.druid.client.ImmutableDruidServer;
-import org.apache.druid.java.util.common.logger.Logger;
 import org.apache.druid.timeline.DataSegment;
 import org.apache.druid.timeline.SegmentId;
 
 import java.util.Objects;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
 
 /**
+ *
  */
 public class ServerHolder implements Comparable<ServerHolder>
```

Comment:
Change summary:
Maintain a map containing
- segments that were in load or drop queue when this run started
- segments that were queued for load or drop during this run
This helps the `SegmentLoader` determine the current state of a segment on a server and make load/drop decisions.